### PR TITLE
Fix/rocksdb: Switched back to main repo

### PR DESCRIPTION
**Pull request by kwek20**
# Description of change

For cross compiling armv7, we need to switch back to the official rocksdb rust crate.
Our fork does not work.

Required for PR #782

## Type of change

- Bug fix (a non-breaking change which fixes an issue)

## How the change has been tested

Cross compiled it :) Main crate builds still

### CODE REVIEW

File: Cargo.toml
Comment (rajivshah3) [inline Cargo.toml]: I think we'll still need to use the Git repo since they haven't included https://github.com/rust-rocksdb/rust-rocksdb/commit/e2e8834975cc9f2cef63ae85d7ea376e7cdc816d in a release yet

```suggestion
rocksdb = { git = "https://github.com/rust-rocksdb/rust-rocksdb", rev = "f98e432623f339d55f7f3cb467dcca6dda62567a", default-features = false, features = ["lz4"] }
```

**Comment by kwek20** [comment]
AH i suppose youre right @rajivshah3 
is there a test we can run to validate?

**Comment by kwek20** [comment]
@rajivshah3 When can we merge this? Id like to get android merged afterwards, and then java docs(wip atm)

**Comment by rajivshah3** [comment]
@Thoralf-M @lmy441900 how should we test these changes?

**Comment by jyhi** [comment]
It's hard to get things right when cross-compiling. @kwek20 could you provide a guide on how to perform the cross-compilation?

**Comment by kwek20** [comment]
The quickest guide would be manual like so: @lmy441900 

```
rustup target add armv7-linux-androideabi

// Export android variables. Set to whatever NDK/android studio you are using
export ANDROID_HOME=/home/brord/Android    (This contains ndk, patcher, platform, build-tools, etc...)
export ANDROID_NDK_HOME=$ANDROID_HOME/ndk/21.1.6352462   (number can be different, i tested on this one)

// Add the standalone toolchain to the search path. Could be windows instead of linux depending on your OS
export PATH=$PATH:$ANDROID_NDK_HOME/toolchains/llvm/prebuilt/linux-x86_64/bin

// Create or update  the following file: "~/.cargo/config", updating path to the same as your $ANDROID_NDK_HOME (Must be full path)
--------
[target.armv7-linux-androideabi]
linker = "/home/brord/Android/ndk/21.1.6352462/toolchains/llvm/prebuilt/linux-x86_64/bin/armv7a-linux-androideabi21-clang"
ar = "/home/brord/Android/ndk/21.1.6352462/toolchains/llvm/prebuilt/linux-x86_64/bin/arm-linux-androideabi-ar"
--------
```

Then run inside `bindings/java/native`:
`BINDGEN_EXTRA_CLANG_ARGS="--target=armv7-linux-androideabi" cargo build --target armv7-linux-androideabi --release`

**Comment by kwek20** [comment]
@rajivshah3 @lmy441900 Can we please get this merged, and the android PR? Ive a doc pr coming up after it already as well but ill hold off on that one until its all merged

**Comment by jyhi** [comment]
@Thoralf-M You decide. I haven't run a test (I didn't managed to setup a correct cross-compilation environment); I choose to trust.

**Comment by Thoralf-M** [comment]
If it works for you and the workflows also pass on all targets I don't see a reason not to update

**Review by Thoralf-M — APPROVED** [review]

**Pull request merged**
